File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.py
def solution(numbers):
    answer = set()
    
    def dfs(st, cur):
        if len(cur) == 2:
            answer.add(sum(cur))
            return
        
        for i in range(st, len(numbers)):
            cur.append(numbers[i])
            dfs(i+1,cur)
            cur.pop()
    
    dfs(0,[])
    return sorted(list(answer))

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    Set<Integer> s = new HashSet<>();
    public void dfs(int[] numbers, Set<Integer> cur) {
        if (cur.size() == 2) {
            int sum = 0;
            for(int x: cur) sum+= numbers[x];
            s.add(sum);
            return;
        }
        
        for(int i=0;i<numbers.length;i++) {
            if(!cur.contains(i)) {
                cur.add(i);
                dfs(numbers,cur);
                cur.remove(i);
            }
        }
        
    }
    public int[] solution(int[] numbers) {
        dfs(numbers, new HashSet<>());
        
        int[] answer = new int[s.size()];
        int idx = 0;
        for(int x: s) answer[idx++] = x;
        Arrays.sort(answer);
        return answer;
    }
}